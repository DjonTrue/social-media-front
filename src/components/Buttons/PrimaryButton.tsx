import React from "react";
import { Button, makeStyles } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  root: {
    margin: theme.spacing(3, 0, 2),
  },
}));

// @ts-ignore
const PrimaryButton = ({ children, props }) => {
  const styles = useStyles();

  return (
    <Button
      className={styles.root}
      type="submit"
      variant="contained"
      color="primary"
      fullWidth
    ></Button>
  );
};

export default PrimaryButton;
