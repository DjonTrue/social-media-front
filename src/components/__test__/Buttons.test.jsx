import React from "react";
import { render, screen } from "@testing-library/react";
import PrimaryButton from "../Buttons/PrimaryButton";

describe("Test PrimaryButton component", () => {
  it("PrimaryButton component renders fine", () => {
    render(<PrimaryButton />);
    const btn = screen.getByRole("button");
    const span = btn.querySelector("span");

    expect(btn).toBeInTheDocument();
    expect(btn).toMatchSnapshot();
    expect(span).toHaveClass("MuiButton-label");
  });
});
