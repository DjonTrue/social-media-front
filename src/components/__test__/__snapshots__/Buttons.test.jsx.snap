// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton component PrimaryButton component renders fine 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-1 MuiButton-containedPrimary MuiButton-fullWidth"
  tabindex="0"
  type="submit"
>
  <span
    class="MuiButton-label"
  />
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;

exports[`Test PrimaryButton component PrimaryButton component renders fine 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-1 MuiButton-containedPrimary MuiButton-fullWidth"
  tabindex="0"
  type="submit"
>
  <span
    class="MuiButton-label"
  />
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;
